src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,50 +1,194 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String EXCEPTION_MESSAGE = "Illegal argument";
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+    }
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.next = next;
+            this.element = element;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addElement(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(EXCEPTION_MESSAGE);
+        }
+        if (index == size) {
+            addElement(value);
+        } else {
+            addElementByIndex(value, index);
+        }
+    }
 
+    private void addElement(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        Node<T> lastForThisNode = last;
+        last = newNode;
+        if (lastForThisNode == null) {
+            first = newNode;
+        } else {
+            lastForThisNode.next = newNode;
+        }
+        size++;
+    }
+
+    private void addElementByIndex(T value, int index) {
+        if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, findByIndex(index));
+            findByIndex(index).prev = newNode;
+            first = newNode;
+            size++;
+        } else {
+            Node<T> after = findByIndex(index);
+            Node<T> before = after.prev;
+            Node<T> newNode = new Node<>(before, value, after);
+            after.prev = newNode;
```

Comment:
this code is the same as the code in the `if` statement, so let's move 4 lines (64-67) before if-else statement

Reply:
done